Modules/ShuttleTrack/MITShuttleRoutesDataSource.m

```diff
@@ -4,69 +4,39 @@
 #import "MITCoreData.h"
 #import "MITAdditions.h"
 #import "MITShuttleRoute.h"
-#import "MITShuttlePredictionList.h"
 #import "MITShuttleController.h"
 
-
-typedef void(^MITShuttleRoutesDataSourceCompletionBlock)(MITShuttleRoutesDataSource *dataSource, NSError *error);
-
-static NSString* const MITMobileErrorOwnerDeallocatedErrorFormat = @"block execution of %p canceled, owner was prematurely deallocated";
-static NSUInteger const MITMobileOwnerDeallocatedError = -16000;
-
-static NSError* MITDispatcherDeallocatedError(void *block) {
-    NSString *errorString = [NSString stringWithFormat:MITMobileErrorOwnerDeallocatedErrorFormat, block];
-    NSError *error = [NSError errorWithDomain:MITErrorDomain
-                                         code:MITMobileOwnerDeallocatedError
-                                     userInfo:@{NSLocalizedDescriptionKey : errorString}];
-
-    DDLogCWarn(@"%@",errorString);
-    return error;
-}
-
 @interface MITShuttleRoutesDataSource ()
-@property(nonatomic,readonly) BOOL didFetchResults;
-@property(nonatomic,strong) NSDate *fetchDate;
-@property(nonatomic,strong) NSError *lastRequestError;
-
-@property(nonatomic,strong) NSManagedObjectContext *managedObjectContext;
-@property(nonatomic,strong) NSFetchedResultsController *fetchedResultsController;
-@property(nonatomic,strong) NSOperationQueue *completionQueue;
-@property(nonatomic,getter=isRequestActive) BOOL requestActive;
 
 @end
 
 @implementation MITShuttleRoutesDataSource
-- (instancetype)init
-{
-    NSManagedObjectContext *managedObjectContext = [[MITCoreDataController defaultController] newManagedObjectContextWithConcurrencyType:NSPrivateQueueConcurrencyType trackChanges:YES];
-    return [self initWithManagedObjectContext:managedObjectContext];
-}
 
-- (instancetype)initWithManagedObjectContext:(NSManagedObjectContext*)managedObjectContext
+- (NSFetchRequest *)fetchRequest
 {
-    self = [super init];
-
-    if (self) {
-        _managedObjectContext = managedObjectContext;
-        _expiryInterval = 300.;
-        _completionQueue = [[NSOperationQueue alloc] init];
-        _completionQueue.maxConcurrentOperationCount = NSOperationQueueDefaultMaxConcurrentOperationCount;
-    }
-
-    return self;
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:[MITShuttleRoute entityName]];
+    fetchRequest.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"order" ascending:YES]];
+    return fetchRequest;
 }
 
-- (instancetype)copyWithZone:(NSZone *)zone
+- (NSArray*)routes
 {
-    MITShuttleRoutesDataSource *newDataSource = [[[self class] alloc] init];
-    newDataSource.fetchDate = self.fetchDate;
-    [newDataSource _performFetch:nil];
-
-    return newDataSource;
+    NSManagedObjectContext *managedObjectContext = [[MITCoreDataController defaultController] mainQueueContext];
```

Comment:
Should this be inside the if statement? Not sure how expensive this operation is and if it really matters.